CoreScriptsRoot/Modules/BackpackScript3D.lua

```diff
@@ -104,10 +148,25 @@ local function AddTool(tool)
 	UpdateLayout()
 end
 
+local humanoidChangedEvent = nil
+local function RegisterHumanoid(humanoid)
+	if humanoidChangedEvent then
+		humanoidChangedEvent:disconnect()
+		humanoidChangedEvent = nil
+	end
+	if humanoid then
+		humanoidChangedEvent = humanoid.Changed:connect(function() UpdateHealth(humanoid) end)
```

Comment:
There is also a special event for just health changes events on humanoid: `humanoid.HealthChanged`
This way we don't have to do a bunch of updates for when other properties change